### ISSUE TITLE
TA Grading collaboration tools

### ISSUE THREAD

**Issue by copperwater** [Submitty/Submitty#1710]
TA grading currently is open to edit wars if two people have different opinions about the rubric and are grading at the same time. There are ways to address this administratively, such as making all graders sit down and agree on a rubric before starting, but there are logistical problems with such an approach (e.g. finding reasons to edit the rubric once grading actually starts).

Two things that would be particularly nice to have:
* A plain text field like the General Comment field but visible only to graders and separate for each question, which can be used by graders to document their reasons for changing marks. This can be done with a shared document outside Submitty, but would be nice if it were directly on the question.
* An alert that the grader is trying to save a new version of the rubric when someone else has already changed it since the grader loaded the page. Perhaps this could be accomplished via having the browser save the version of the rubric that was loaded with the page (whether uncompressed or hashed) and comparing it at save time to the version stored in the database. If these versions differ, it should then prevent the save and force the grader to refresh the page to get the new rubric.

Some of this is already covered in #1500, but I don't know to what extent.

**Comment by bmcutler** [comment]
use case of private thread/channel of discussion forum for TAs

**Comment by bmcutler** [comment]
This idea was captured in the TA Notes/Solutions tab of TA grading.

**Comment by copperwater** [comment]
Good to see that grading continues to be made smoother!
